nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -391,7 +394,12 @@ def save_checkpoint(
             sharded_optim_state = self.optimizer_sharded_state_dict(
                 unsharded_optim_state=checkpoint['optimizer_states'][0]
             )
-            checkpoint['optimizer_states'] = [sharded_optim_state]
+
+            # Check whether to save optim states
+            if storage_options.get('drop_optim_states', False):
```

Comment:
Even though these are different strategies, maybe let's decide on a single name to reduce confusion, see parallel PR for NeMo 2.0: https://github.com/NVIDIA/NeMo/pull/10241/files#diff-78f81f4094cfea056c177e87c0d527b9ce27cee11813138e5a2a69370b922c19R610
I'm OK with both.

cc @ashors1

Reply:
okay, I'm fine with the name @ashors1 used. Not a big deal I think.